satellite/metabase/rangedloop/service.go

```diff
@@ -52,20 +53,27 @@ func NewService(log *zap.Logger, config Config, provider RangeSplitter, observer
 }
 
 // observerState contains information to manage an observer during a loop iteration.
-// Improvement: track duration.
 type observerState struct {
 	observer       Observer
 	rangeObservers []*rangeObserverState
+	// err is the error that occurred during the observer's Start method.
+	// If err is set, the observer will be skipped during the loop iteration.
+	err error
 }
 
 type rangeObserverState struct {
 	rangeObserver Partial
 	duration      time.Duration
+	// err is the error that is returned by the observer's Fork or Process method.
+	// If err is set, the range observer will be skipped during the loop iteration.
+	err error
 }
 
 // ObserverDuration reports back on how long it took the observer to process all the segments.
 type ObserverDuration struct {
 	Observer Observer
+	// When this observer errored out, set duration to -1s
```

Comment:
nitpick: comment should start with the field name, i.e. `Duration`

Reply:
I'll sneak that in another change so I don't lose the approve.

Reply:
Done in https://review.dev.storj.io/c/storj/storj/+/9156